packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/UserImpl.kt

```diff
@@ -94,6 +94,7 @@ public class UserImpl(
         }
 
     override suspend fun logOut() {
+        val reportLoggedOut = loggedIn
```

Comment:
This is not really safe. Two threads calling this simultaneous would issue two events but should only emit one. I guess we would either need a return value from core or synchronize the operations for a use. Don't know if it is a critical issue though. Same for the other events.

Reply:
Good catch. I already created https://github.com/realm/realm-core/issues/6516 which will also fix this. I'll think and see if there is something we can do to prevent this, but if it is super tricky I would probably just ignore this case. I doubt this is code that will see race conditions anyway.

Reply:
I reworked the logic a little bit. The chance for a race condition should be less now, although still present. My stance would probably be that using locks for this would be overkill and we should wait for https://github.com/realm/realm-core/issues/6516 for a proper fix, but not sure what you think?

Reply:
Sounds ok to leave it as is for now. Maybe include a link to the core issue in a comment  above `reportLoggedOut`.